ui/src/app/views/workspace-edit/component.ts

```diff
@@ -53,6 +50,9 @@ export class WorkspaceEditComponent implements OnInit {
       }};
     if (this.route.routeConfig.data.adding) {
       this.adding = true;
+      this.errorHandlingService.retryApi(this.profileService.getMe()).subscribe(profile => {
```

Comment:
This will get run every time a user creates a workspace, right? What happens if this hasn't finished before the user hits submit?

While this isn't the end of the world, we really only need to fetch this info once (it won't change.) Should we consider caching it in some application state somewhere on login?

Alternatively we could have a protocol where if the client doesn't indicate what namespace to use, we default to their free tier billing project.

Reply:
Yes, this is terrible, and yet still better than what we currently have, which is a hard-coded project that probably should no longer exist. I'd like to get this entire flow working before we worry about making it not suck.

Reply:
Fair enough!

Reply:
(Though please file a JIRA ticket so we don't forget.)

Reply:
https://precisionmedicineinitiative.atlassian.net/browse/RW-230